### ISSUE TITLE
Support JS style line comments (// comment)

### ISSUE THREAD

**Issue by wachunga** [styled-components/stylelint-processor-styled-components#156]
For example:
```js
import styled from 'styled-components';
export default styled.div`
  width: 100%; // fails
`;
```

Converting to a typical CSS comment `/* */` avoids the error. But line comments are convenient, and my understanding is that styled-components supports them.

```
├─ stylelint-config-recommended@2.0.0
├─ stylelint-config-standard@18.0.0
├─ stylelint-config-styled-components@0.1.1
├─ stylelint-processor-styled-components@1.2.1
└─ stylelint@8.3.1
```

**Comment by wachunga** [comment]
If this is intended behaviour, it could at least fail more gracefully. As it is, this error obscures all other lint warnings and reports incorrect line numbers. Took me a while to figure out the issue.

**Comment by emilgoldsmith** [comment]
Hmm if you're right that Styled Components supports line comments (I'm not 100% sure on that one) then that's a good point.

The incorrect line numbers is in the process of being fixed #154 but we need to wait for a PR to be merged on the Stylelint side.

The tricky part about this is that all we do is pass css to Stylelint and they do the rest of the work (and the error you encountered), but I guess we could for example parse the css and manually remove all line comments before we send the css to Stylelint.

If someone like @mxstbr could just confirm that Styled Components indeed supports line comments that'd be great.

**Comment by wachunga** [comment]
Thanks for the prompt reply. See https://github.com/styled-components/styled-components/issues/1098#issuecomment-324404085, where 'supper' I believe means 'support'.

**Comment by emilgoldsmith** [comment]
Yep I can see this test as well: https://github.com/styled-components/styled-components/blob/8aa30c61818ae4eae4f7266b49923162e1b68170/src/test/styles.test.js#L164-L178

**Comment by emilgoldsmith** [comment]
So yep, this is officially verified, and is a good enhancement to stay consistent with Styled Components. I definitely think the way to go is stripping the comments, I would say go look at the implementation done of the stripping in Styled Components and just copy that for consistency, though I think it should be pretty simple as long as we only operate on the final CSS (I'm not sure if you could run into some edge cases with comments in and outside of interpolations? Like `// ${ something }` would actually be parsed as an interpolation by JS as JS doesn't see the comment inside the string).

**Comment by emilgoldsmith** [comment]
Hey @wachunga, I was just trying to look at this, and wrote some tests and they all pass... Do you still experience these problems with the newest versions of everything? If so could you maybe submit a PR with a breaking test case? I could also submit a PR with the test cases I have so you could add to them if you want

**Comment by wachunga** [comment]
@emilgoldsmith I'm still getting the error after upgrading to `1.2.2` and stylelint `8.4.0`:

```
severity: 'Error'
message: 'Unknown word (CssSyntaxError)'
at: '3,23'
source: 'stylelint'
code: 'undefined'
```
I think submitting your test cases makes sense. 👍 

```
─ stylelint-config-standard@^18.0.0
─ stylelint-config-styled-components@^0.1.1
─ stylelint-processor-styled-components@^1.2.2
─ stylelint@^8.4.0
```

**Comment by emilgoldsmith** [comment]
Done, see #164

**Comment by emilgoldsmith** [comment]
So @wachunga if you could please help troubleshoot what it is that doesn't work on your machine that seems to do on mine (assuming the CI passes the tests which I think it will as they pass locally for me), that'd be great.

**Comment by emilgoldsmith** [comment]
@wachunga the tests pass so I'm not quite sure what your problem is, as I said above you're going to have to help troubleshoot it if you keep having issues

**Comment by emilgoldsmith** [comment]
Are you using `syntax: scss` in your stylelint config? It is stated in the README that that is necessary for Styled Components to lint properly

**Comment by chinesedfan** [comment]
@emilgoldsmith Hi, what do you mean `It is stated in the README`? I didn't find it.

**Comment by emilgoldsmith** [comment]
Oh this is no longer relevant as of #170

**Comment by kud** [comment]
I'm not sure to undersand everything.

I've got this: 

```
const Root = styled.div`
  background-size: cover;
  // background-image: linear-gradient(225.84deg, #6EBCCD -3.01%, #4787BF 67.2%), url("http://placeimg.com/1440/600/arch");
  background-image: url(${heroImageSrc});
  background-blend-mode: color;

  width: 100%;
  height: 600px;
`
```

and due to my JS comment, I've got a CssSyntaxError. What should I do? Just adding the syntax: scss? Sounds to be a strong solution of just this...?
